### ISSUE TITLE
Polarized SANS metadata: SaveNXcanSAS should accept workspace group inputs

### ISSUE THREAD

**Issue by rbauststfc** [mantidproject/mantid#38504]
Part of #36147. See the design and requirements documents that are linked on that issue for more context.

The `SaveNXcanSAS` algorithm will only take a Matrix Workspace, however the output of a polarized SANS reduction will be a workspace group. In the code, although we plan to add a `SavePolarizedNXcanSAS` algorithm for reduced polarized data, I'm expecting that we will copy the `InputWorkspace` property for that algorithm from `SaveNXcanSAS`. This means we will need to change the `InputWorkspace` property on `SaveNXcanSAS` so that it will accept workspace groups, or potentially a list of workspace names (see final paragraph).

At the moment the SANS scientists save workspace groups from non-polarized reductions (e.g. when time slicing) into separate files, but this is much less convenient for them. They would like to be able to pass a workspace group to `SaveNXcanSAS` and have it save all of the data into a single file. This would involve writing multiple `SASData` blocks into the file and I'm not sure what metadata our scientists would want us to add to distinguish each block. This is outside scope for the Polarised SANS epic, so I would recommend that we simply allow workspace groups to be passed in but internally still save them into separate files. Following further discussion with our scientists, a separate issue can be opened to look at saving multiple `SASData` blocks into a single file from `SaveNXcanSAS`.

It's worth taking a look at issue #38525 before starting this/concluding what's needed here. That issue discusses the changes that will eventually be needed to integrate the save step into the SANS reduction workflow. It mentions that we'll likely need to refactor/extend the existing save steps because currently a list of workspace names is saved out as individual files. It's worth thinking ahead to what we might need to be able to pass in from the reduction code in order to determine exactly how to approach this.

**_This issue should only be worked on once #38503 has been completed._**
